apps/extension/src/ui/apps/dashboard/routes/Settings/AssetsDiscovery/AssetDiscoveryPage.tsx

```diff
@@ -167,13 +166,10 @@ const AssetRowContent: FC<{ tokenId: TokenId; assets: DiscoveredBalance[] }> = (
     (e) => {
       const checked = e.target.checked
       if (!token || !evmNetwork) return
-      // when unchecking, dont disable the network unless it was originally disabled, except for native tokens
-      if (
-        checked ||
-        (!checked && !originalActiveEvmNetworks.current[evmNetwork.id]) ||
-        token.type === "evm-native"
-      )
-        activeEvmNetworksStore.setActive(evmNetwork.id, checked)
+
+      if (checked) activeEvmNetworksStore.setActive(evmNetwork.id, true)
+      // when unchecking, dont disable the network except for native tokens
+      else if (token.type === "evm-native") activeEvmNetworksStore.setActive(evmNetwork.id, false)
```

Comment:
I find this easier to read:

```suggestion
      if (!checked && token.type === "evm-native") activeEvmNetworksStore.setActive(evmNetwork.id, false)
```